### DISCUSSION TITLE
Sharing web worker between multiple monaco language client

### DISCUSSION THREAD

**Discussion by wa8f7192** [TypeFox/monaco-languageclient#663] [Q&A]
In the example given https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/src/langium/statemachine/main.ts

It creates 2 wrappers, where each wrapper is passed in a different web worker,

Would it be feasible to use only 1 worker and share between wrappers? and whether there are concerns on doing this.

Assuming we will only show one editor at a time.

**Comment by kaisalmen** [comment]
Hi @wa8f7192 this is the proposed solution for #620 (see PR #666) and yes that is viable solution if both editors use the same language(Id).

**Reply by yangerxia** [reply]
This dependency was not found：
vscode/services in ./node_modules/monaco-languageclient/lib/monaco-services.js
how to solve problems？

**Reply by CGNonofr** [reply]
Your problem seems completely unrelated, please open your own issue or discussion
